src/module-elasticsuite-virtual-category/Model/Rule.php

```diff
@@ -181,9 +181,9 @@ public function getSearchQueriesByChildren(CategoryInterface $rootCategory): arr
                 ->addIsActiveFilter()
                 ->addIdFilter($childrenIds)
                 ->addAttributeToSelect(['virtual_category_root', 'is_virtual_category', 'virtual_rule']);
-
+            $excludedCategories = [];
             foreach ($categoryCollection as $category) {
-                $childQuery = $this->getCategorySearchQuery($category);
+                $childQuery = $this->getCategorySearchQuery($category, $excludedCategories);
```

Comment:
Initializing/resetting the `$excludedCategories` should be done here.

=> This method is called to build the category facet/aggregation : each \Smile\ElasticsuiteVirtualCategory\Model\Rule::getCategorySearchQuery should therefore be independent, otherwise children virtual categories sharing the same virtual root category could simply "disappear" from the "categories" aggregation.